src/infrastructure/s3/s3Service.ts

```diff
@@ -0,0 +1,34 @@
+import { S3 } from 'aws-sdk';
+import IncorrectFileTypeError from '../../errors/IncorrectFileTypeError';
+import NoBodyError from '../../errors/NoBodyError';
+
+export default async (
+  s3: S3,
+  bucket: string,
+  folder: string | undefined,
+  certId: string,
+  vin: string,
+): Promise<S3.Body> => {
+  const key = folder ? `${folder}/${certId}_${vin}.pdf` : `${certId}_${vin}.pdf`;
+
+  console.info(`Bucket name: ${bucket}`);
+  console.info(`Item key: ${key}`);
+
+  const response = await s3
+    .getObject({
+      Bucket: bucket,
+      Key: key,
+    })
+    .promise();
+
+  if (response.ContentType !== 'application/octet-stream') {
+    console.error(`Incorrect content-type: ${response.ContentType}`);
+    throw new IncorrectFileTypeError();
+  }
+
+  if (response.Body) {
```

Comment:
do we want to wrap the await in try catch block so if `.getObject()` fails what do we do?
This will most likely cause a silent error hard to trace should things go wrong or we saying that whatever happen with this async operation we default to the NoBodyError() call ?